cocos/core/event/callbacks-invoker.ts

```diff
@@ -174,13 +208,16 @@ export class CallbacksInvoker {
      * @param once - Whether invoke the callback only once (and remove it)
      */
     public on (key: string, callback: Function, target?: Object, once?: boolean) {
-        let list = this._callbackTable[key];
-        if (!list) {
-            list = this._callbackTable[key] = callbackListPool.alloc();
+        if (!this.hasEventListener(key, callback, target)) {
+            let list = this._callbackTable[key];
+            if (!list) {
+                list = this._callbackTable[key] = callbackListPool.alloc();
```

Comment:
BTW, I saw the renaming of pool's API. I don't think alloc/free are better than get/put. The former seems like memory operation, the latter describes more precisely for pool operations.

Reply:
Won't be modified in this PR